packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -206,18 +204,76 @@ class GuildMemberManager extends CachedManager {
    */
   fetch(options) {
     if (!options) return this._fetchMany();
-    const user = this.client.users.resolveId(options);
-    if (user) return this._fetchSingle({ user, cache: true });
-    if (options.user) {
-      if (Array.isArray(options.user)) {
-        options.user = options.user.map(u => this.client.users.resolveId(u));
-        return this._fetchMany(options);
-      } else {
-        options.user = this.client.users.resolveId(options.user);
-      }
-      if (!options.limit && !options.withPresences) return this._fetchSingle(options);
+    const { user, cache, force } = options;
+
+    const {
+      client: { users },
+    } = this;
+
+    const resolvedUser = users.resolveId(user ?? options);
+    if (resolvedUser) return this._fetchSingle({ user: resolvedUser, cache, force });
+    const resolvedUsers = user?.map(_user => users.resolveId(_user));
```

Comment:
If we don't destructure `users` out of `client`, we can give the variable a better name: `users`.

Also, this is quite assuming that if it couldn't resolve a userId, it's then an array. That won't translate well in TypeScript. 👀

Reply:
I've removed the destructure with 207289fabb8067d1fb26a490566e42c95fbdc86f.

> Also, this is quite assuming that if it couldn't resolve a userId, it's then an array. That won't translate well in TypeScript. 👀

To be honest, this method performs a gateway request and a REST GET for a single guild member. To do both, `user` is either a single user or an array of users. It'd probably be simpler in the future to only have an array... or I was thinking to separate these two functionalities into their own methods.